### PR TITLE
Improves native emoji conversion

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js.es6 [hunk 1]
```diff
@@ -7,7 +7,7 @@ let _unicodeRegexp;
 export function setUnicodeReplacements(replacements) {
   _unicodeReplacements = replacements;
   if (replacements) {
-    _unicodeRegexp = new RegExp(Object.keys(replacements).join("|"), "g");
+    _unicodeRegexp = new RegExp(Object.keys(replacements).sort().reverse().join("|"), "g");
```
Comment (ZogStriP) [inline app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js.es6]: Why the `.reverse()`?

Comment (jjaffeux) [inline app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js.es6]: we want the longer emojis sequences to be match first:

👩🏾‍🎓  before 👩🏾‍  before 👩

does it make sense ?

Comment (ZogStriP) [inline app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js.es6]: It does. Might be worth adding a comment in the code

Comment (jjaffeux) [inline app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js.es6]: https://github.com/discourse/discourse/pull/4930

Thanks for feedback

**Pull request merged**